userspace/falco/app_actions/process_events.cpp

```diff
@@ -189,7 +189,9 @@ application::run_result application::do_inspect(
 			return run_result::fatal("Drop manager internal error");
 		}
 
-		if(!ev->simple_consumer_consider() && !m_options.all_events)
+		/* If we have not set the `-A` flag and the event is unused or old, interrupt the flow */
+		uint16_t evt_type = ev->get_type();
+		if(!m_options.all_events && (sinsp::is_unused_event(evt_type) || sinsp::is_old_version_event(evt_type)))
```

Comment:
Not sure about the utility of this check... here we check for an old version event or for an unused event and we interrupt the flow, but this should be always true also if the `-A` flag is enabled :thinking:

Reply:
I agree that having this check now is a bit weird because it is always true, or we are saying sinsp is not consistent.
Could we introduce a simple regression test to be relatively sure of our code without checking the same logic at runtime twice?

(If it were too late for this release, I would propose to leave it as-is and  `todo:`).

Reply:
my question here is why we are dropping old version events?

Reply:
I think that this check is no more useful I will remove it, unused event should be filtered by sinsp and old events should be considered if received and not dropped